assets/javascripts/src/modules/abTests.es6

```diff
@@ -21,27 +19,94 @@ export function init() {
     }
 }
 
-function abTestData(tests, complete) {
+/**
+ * Check if an object is empty.
+ * @see http://stackoverflow.com/questions/679915/how-do-i-test-for-an-empty-javascript-object
+ */
+function isEmpty(obj) {
+    return Object.keys(obj).length === 0 && obj.constructor === Object;
+}
+
+/**
+ * The function f should take a test and return the complete flag (true/false) for the test.
+ * If the function returns null, the complete flag for the test will not be set.
+ * Any tests which have their complete flag set are registered with Ophan.
+ */
+function registerTestsWithOphan(tests, f) {
     const data = {};
 
     for (var test of tests) {
-        data[test.testSlug] = {
-            'variantName': test.variantSlug,
-            'complete': String(complete)
+        const completeFlag = f(test);
+
+        if (completeFlag !== null) {
+            data[test.testSlug] = {
+                'variantName': test.variantSlug,
+                'complete': String(completeFlag)
+            }
         }
     }
 
-    return data;
+    if (!isEmpty(data)) {
+        ophan.loaded.then(function (ophan) {
+            ophan.record({
+                abTestRegister: data
+            })
+        })
+    }
+}
+
+/**
+ * @see AARecurringTest for the corresponding Scala class.
+ */
+function isAARecurringTest(test) {
+    // Only one variant name to check.
+    return (test.testName == 'AARecurringTest') && (test.variantName == 'default');
+}
+
+/**
+ * To be used with registerTestsWithOphan().
+ * Sets a test's completed flag to false, unless it is the AA recurring test,
+ * in which case its complete flag is set to true.
+ */
+function initTestFlags(test) {
+   return isAARecurringTest(test);
+}
+
+/**
+ * To be used with registerTestsWithOphan().
+ * Set a test's complete flag to true, apart from the AA recurring test which is ignored.
+ */
+function completeTestFlags(test) {
+    if (isAARecurringTest(test)) {
+        return null;
```

Comment:
My idea for this test was to fire `true` straight away so we should see as many `falses` as there are `true`s